### PR TITLE
Add/language picker suggested languages

### PR DESCRIPTION
**Pull request by ramonjd**
This PR is a continuation of @jsnajdr's work in #13493.

I have rebased, added tests, performed a bit more browser testing, and offered my propitiations to the linting demons.

We're presenting a list of suggested languages based the languages a user has defined in his or her browser settings. If the feature is missing from the browser, we don't show the suggested list, for example, Windows/Internet Explorer [does not yet support](https://developer.mozilla.org/en-US/docs/Web/API/NavigatorLanguage/languages) `window.navigator.languages`, the property we're relying on for this feature.

suggested languages

## Testing instructions:

1. In your browser's preferences, add a few languages, making sure they are supported by WordPress.com.
2. Go to `/me/account`

### Expectations
- "Suggested languages" list should reflect the browser preferences correctly
- Selecting a suggested language will also select the corresponding language in the list

### CODE REVIEW

**Comment by folletto** [comment]
Design wise, works well and looks good. 👍

File: client/components/language-picker/modal.jsx [hunk 1]
```diff
@@ -140,12 +177,32 @@ class LanguagePickerModal extends PureComponent {
...
+		const { suggestedLanguages } = this.state;
+		if ( ! suggestedLanguages ) {
+			return null;
+		}
+
+		return (
+			<div className="language-picker__modal-suggested">
+				<div className="language-picker__modal-suggested-inner">
+					<div className="language-picker__modal-suggested-label">
+						{ this.props.translate( 'Suggested languages: ' ) }
```
Comment (akirk) [inline client/components/language-picker/modal.jsx]: The trailing space is an indicator of string concattenation. Can we add the languages as a component in the translation? If not, please remove the whitespace and the translators will accommodate as good as possible.

File: client/components/language-picker/test/modal.js [hunk 2]
```diff
@@ -0,0 +1,90 @@
...
+	};
+
+	test( 'should render', () => {
+		const wrapper = shallow( <LanguagePickerModal { ...defaultProps } /> );
+
+		expect( wrapper ).toMatchSnapshot();
+	} );
+
+	describe( 'suggested languages', () => {
+		const browserLanguages = [ 'en-AU', 'en', 'en-US', 'it' ];
```
Comment (akirk) [inline client/components/language-picker/test/modal.js]: Could we test more combinations here? For example from the code above I think `[ 'en-US', 'it' ]` would only suggest Italian. Let's try even more:
```
[],
[ 'en-US' ],
[ 'en' ],
[ 'it' ],
[ 'it', 'en-US' ],
[ 'it', 'en' ],
[ 'cs', 'it', 'en-US' ]
```

**Comment by akirk** [comment]
Tested it and it works well. Great!

File: client/components/language-picker/test/modal.js (hunk 2 above)
Comment (ramonjd) [reply client/components/language-picker/test/modal.js]: We're also testing that we only show supported languages, regardless of the items in browser languages. On that note, I'll expand the mocks so we can extend the tests further. 👍

File: client/components/language-picker/modal.jsx (hunk 1 above)
Comment (ramonjd) [reply client/components/language-picker/modal.jsx]: Unfortunately there's a lot of cruft around the language texts. I'll remove the white space for now.

**Pull request merged**